contracts/MetaStep.sol

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 Cartesi Pte. Ltd.
+
+// SPDX-License-Identifier: Apache-2.0
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+/// @title MetaStep
+/// @notice State transition function that takes the machine from micro-state s[i] to s[i + 1]
+
+pragma solidity ^0.8.0;
+
+import "./interfaces/IMetaStep.sol";
+import "./interfaces/IUArchStep.sol";
+import "./AccessLogs.sol";
+import "./UArchConstants.sol";
+import "./UArchState.sol";
+
+contract MetaStep is IMetaStep {
```

Comment:
Yeah I can try.

Reply:
So I turned everything into libraries except `UArchState` and `MetaStep`. Mainly because we need the flexibility of using different `AccessLogs` for testing, and not exposing this issue to regular users as described in #43